test.js

```diff
@@ -244,3 +246,134 @@ test('sync (not found, custom cwd)', t => {
 
 	t.is(filePath, null);
 });
+
+test('async (matcher function)', async t => {
+	const cwd = process.cwd();
+
+	t.is(await m(dir => {
```

Comment:
Can you use the `directory` wording for all these instead of `dir`?

Reply:
Done.